srtcore/core.cpp

```diff
@@ -7901,61 +7890,60 @@ int CUDT::processData(CUnit* unit)
           HLOGC(dlog.Debug, log << "crypter: data not encrypted, returning as plain");
       }
 
-   }  /* End of recvbuf_acklock*/
+   }  /* End of recvbuf_acklock */
 
-   if (m_bClosing) {
-      /*
-      * RcvQueue worker thread can call processData while closing (or close while processData)
-      * This race condition exists in the UDT design but the protection against TsbPd thread
-      * (with AckLock) and decryption enlarged the probability window.
-      * Application can crash deep in decrypt stack since crypto context is deleted in close.
-      * RcvQueue worker thread will not necessarily be deleted with this connection as it can be
-      * used by others (socket multiplexer).
-      */
+   if (m_bClosing)
+   {
+      // RcvQueue worker thread can call processData while closing (or close while processData)
+      // This race condition exists in the UDT design but the protection against TsbPd thread
+      // (with AckLock) and decryption enlarged the probability window.
+      // Application can crash deep in decrypt stack since crypto context is deleted in close.
+      // RcvQueue worker thread will not necessarily be deleted with this connection as it can be
+      // used by others (socket multiplexer).
       return(-1);
    }
 
    // If the peer doesn't understand REXMIT flag, send rexmit request
    // always immediately.
    int initial_loss_ttl = 0;
-   if ( m_bPeerRexmitFlag )
+   if (m_bPeerRexmitFlag)
        initial_loss_ttl = m_iReorderTolerance;
 
-   if  (packet.getMsgCryptoFlags())
+   if (packet.getMsgCryptoFlags())
    {
-       /*
-       * Crypto flags not cleared means that decryption failed
-       * Do no ask loss packets retransmission
-       */
+       // Crypto flags not cleared means that decryption failed
+       // Do not ask for lost packets retransmission
        ;
        HLOGC(mglog.Debug, log << CONID() << "ERROR: packet not decrypted, dropping data.");
    }
-   else
-   // Loss detection.
-   if (CSeqNo::seqcmp(packet.m_iSeqNo, CSeqNo::incseq(m_iRcvCurrSeqNo)) > 0)
+   else if (CSeqNo::seqcmp(packet.m_iSeqNo, CSeqNo::incseq(m_iRcvCurrSeqNo)) > 0)
    {
+       // A loss is detected
        {
-           CGuard lg(m_RcvLossLock);
-           int32_t seqlo = CSeqNo::incseq(m_iRcvCurrSeqNo);
-           int32_t seqhi = CSeqNo::decseq(packet.m_iSeqNo);
-           // If loss found, insert them to the receiver loss list
+           // TODO: Can unlock rcvloss after m_pRcvLossList->insert(...)?
```

Comment:
`m_FreshLoss` probably should be also protected. Otherwise it can be unlocked.

Reply:
I would leave it as a TODO, because this PR should not introduce changes, only minor code cleanup. Added you note to the comment.